snap/snapcraft.yaml

```diff
@@ -0,0 +1,134 @@
+name: edgex-device-usb-camera
+base: core20
+license: Apache-2.0
```

Comment:
While the camera service itself is licensed as Apache 2.0, some of the other components in this snap have other licenses (e.g. rtsp-simple-server is MIT).

Also this snap is including ffmpeg which is licensed under LGPLv2, however some components of ffmpeg are licensed under GPLv2 (see https://www.ffmpeg.org/legal.html). Was this reviewed as part of the new service review?

Same question about libpulse0 (which again is licensed under LGPLv2)? The pulseaudio license also mentions that if D-Bus support is enabled, the PulseAudio client library (libpulse) MAY need to be licensed under the GPL, depending on the license adopted for libdbus. Also do we know why service requires libpulse0? Is there a runtime requirement for the PulseAudio server in order to use this service?

Reply:
> While the camera service itself is licensed as Apache 2.0, some of the other components in this snap have other licenses (e.g. rtsp-simple-server is MIT).

This should be fine since MIT is permissive.
I've also considered not including rtsp server in the snap and let users install their own server (e.g. [v4l2-rtspserver](https://snapcraft.io/v4l2-rtspserver)). But in the end, I decided to keep it since it is only 9MB and disabled by default.

> Also this snap is including ffmpeg which is licensed under LGPLv2, however some components of ffmpeg are licensed under GPLv2 (see https://www.ffmpeg.org/legal.html). Was this reviewed as part of the new service review?


I did not review the docker base images and packages, not sure if someone else did. @iain-anderson @lenny-intel an idea?
This would be a project issue and not specific to the snap.

Reply:
According to explicit section in GNU's license agreement (section 5 in the document linked below), we are ok to use the library and still distribute as Apache 2.  https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.  This is known as the linking exception.  The same was applied when we worked with 0MQ.  LF legal cannot provide legal advise (strange but true), but their reading of this also pointed us to this exception.  (see https://en.wikipedia.org/wiki/GPL_linking_exception).

Reply:
@jpwhitemn you're misinterpreting my comment. I acknowledged that the ffmpeg package is licensed as **LGPLv2**, however there are optional parts of ffmpeg that are licensed as **GPLv2**. My question was simply did we review our usage of ffmpeg to ensure we didn't use one of these optional parts?

Reply:
My point is that the snap metadata is saying the license of this snap is "Apache 2.0", and that's not technically correct, as the snap (and most likely our docker image) contains more than just the device service Go binary. The guidance from the snapcraft.io [reference](https://snapcraft.io/docs/snapcraft-yaml-reference) for `snapcraft.yaml `say this needs to be an SPDX license expression (see https://spdx.dev/spdx-specification-21-web-version/), which in this case should be a list of licenses as there's doesn't seem to be an option for "multiple licenses" anymore.



For instance, here are a number of shared libraries that and up in the snap and their associated licenses:

- libpango-1.0.0 - LGPL2
- libvorbisenc2 - BSD
- libxvidcore4 - GPL2+

And also while reviewing the files in the snap, I don't see a license file for the rtsp-server itself.

Reply:
There are close to 200 packages pulled in for ffmpeg and libzmq5. Many of those (incl. libzmq ones) have multiple licenses. It is simply not practical to identify and include all these, and maintain them over time. I don't see any other Docker image or snap listing all licenses.

I found some useful information at:
- https://www.linuxfoundation.org/tools/docker-containers-what-are-the-open-source-licensing-considerations/
- https://opensource.stackexchange.com/a/8393

I believe the license in the Dockerfile is the Dockerfile's license, rather than the license of Docker images built with it. For the snap, the license field appear to be the license of the resulting snap packages. So, IMO, it is most appropriate to remove the optional `license` field. All licenses remain available under snap's `/usr/share/doc` directory.

The license for rtsp-simple-server is at snap's `/usr/share/doc/rtsp-simple-server/LICENSE`.

Reply:
I have remove the license field from the snap: https://github.com/edgexfoundry/device-usb-camera/pull/30/commits/4c4ab6af09efedeff12eed0a9f83e12c8148dad6